main.go

```diff
@@ -118,93 +143,113 @@ type writer interface {
 	Name() string
 }
 
+type noOpWriter struct{}
+
+func (no *noOpWriter) Write(samples model.Samples) error {
+	util.LogDebug("msg", "Noop writer", "num_samples", len(samples))
+	return nil
+}
+
+func (no *noOpWriter) Name() string {
+	return "noopWriter"
+}
+
 type reader interface {
 	Read(req *prompb.ReadRequest) (*prompb.ReadResponse, error)
 	Name() string
 	HealthCheck() error
 }
 
-func buildClients(logger log.Logger, cfg *config) ([]writer, []reader) {
-	var writers []writer
-	var readers []reader
+func buildClients(cfg *config) (writer, reader) {
+	var writer writer
```

Comment:
You can now completely remove these local vars and just do `return pgClient, pgClient`.